crawl-ref/source/describe.cc

```diff
@@ -3884,50 +3884,50 @@ static void _add_energy_to_string(int speed, int energy, string what,
 }
 
 /**
- * Calculate effects on the player's to-hit from a defending monster.
+ * Calculate post-randomization effects on the player's to-hit.
  *
- * @param base_to_hit   The to-hit with only attacker-side effects applied.
+ * @param to_hit        The actual randomized to-hit.
  * @param mi[in]        Player-visible info about the monster in question.
  * @param melee         Whether the attack is melee or ranged.
  */
-static int _apply_defender_effects(const monster_info& mi, int to_hit, bool melee)
+static int _post_roll_to_hit_modifiers(const monster_info& mi, int to_hit, bool melee)
 {
     int effect = 0;
-    // Lighting effects. Subtracting these from EV is weirdly equivalent to adding
-    // them to the post-roll to-hit (the to_hit we have here is pre-roll).
+
+    effect -= 5 * you.inaccuracy();
+
+    if (you.confused()) effect -= 5;
+
+    // Lighting effects.
     if (mi.is(MB_GLOWING)       // corona, silver corona (!)
         || mi.is(MB_BURNING)    // sticky flame
         || mi.is(MB_HALOED))
     {
         effect += BACKLIGHT_TO_HIT_BONUS;
-    } else if (mi.is(MB_UMBRAED) && !you.nightvision())
+    }
+    else if (mi.is(MB_UMBRAED) && !you.nightvision())
         effect += UMBRA_TO_HIT_MALUS;
 
     if (melee) {
-        // Duplicates melee_attack::calc_to_hit().
-
+        // Duplicates melee_attack::post_roll_to_hit_modifiers().
         if (you.duration[DUR_CONFUSING_TOUCH])
-            effect += you.dex() / 2;
+            effect += maybe_random_div(you.dex(), 2, false);
```

Comment:
This is literally `you.dex() / 2`, right?

Reply:
No, it's you.dex()/4. maybe_random_div in random mode puts out a random number that's between 0 and ceil(nom/denom), so half of that is nom/2/denom.

But I figured using the maybe_random_div's built-in non-random mode was good because it allows future devs to easily change what we use as the int approximation of the mean of a distribution. (Notably, the existing code is not actually the nearest int to the actual mean in all cases; sometimes rounding up would be more accurate. But this keeps it standardized, and a change can happen in one place if anyone decides to improve it.)